app/types.ts

```diff
@@ -9,3 +9,30 @@ export type CourseRow = {
   instructorName: string;
   courseTitle: string;
 };
+
+export type SummaryRow = {
+  enrollmentTerm: string;
+  subjectArea: string;
+  catalogNumber: string;
+  sectionNumber: string;
+  enrollmentTotal: string;
+  instructorName: string;
+  courseTitle: string;
+  gradeAverage: number | null;
+  gradeSpread: number | null;
```

Comment:
for this project let's prefer `undefined` to `null` where possible — typescript has cleaner syntax for handling `undefined` IMO

```ts
gradeAverage?: number;

// same as

gradeAverage: number | undefined;
```

(i only use [`null` in `useRef`](https://stackoverflow.com/questions/56541342/react-hooks-why-is-current-null-for-useref-hook) because React *itself* returns `null` from the hook)